### PR TITLE
Use javax.annotation.Nullable instead of androidx.annotation.Nullable  to decouple Android dependency

### PR DESCRIPTION
**Pull request by yufengwangca**
https://github.com/project-chip/connectedhomeip/issues/23811

Currently, androidx.annotation.Nullable is used in Matter Controller Java package, this prevent us from using matter java package in a Java application running on desktop.

Replace androidx.annotation.Nullable with javax.annotation.Nullable

Testing:

Build both Adnroid CHIPTool and java-matter-controller, and running on Android phone and Linux, no regression found.

### CODE REVIEW

**Review by yunhanw-google — APPROVED** [review]

File: third_party/java_deps/BUILD.gn [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# === This BUILD file is auto-generated, modify the GnBuildGenerator task instead of editing this file manually. ===
+
+import("//build_overrides/chip.gni")
+import("${chip_root}/build/chip/java/rules.gni")
+
+java_prebuilt("annotation") {
+  jar_path = "artifacts/jsr305-3.0.2.jar"
```
Comment (yunhanw-google) [inline third_party/java_deps/BUILD.gn:21]: please force check-in jsr305 jar file

Comment (andy31415) [reply third_party/java_deps/BUILD.gn:21]: can we grab it somehow from maven/cipd to not check in binary files.

**Review by andy31415 — APPROVED** [review]

**Comment by gmarcosb** [comment]
LGTM!

**Review by gmarcosb — APPROVED** [review]
LGTM!

File: third_party/java_deps/BUILD.gn (hunk 1 above)
Comment (yufengwangca) [reply third_party/java_deps/BUILD.gn:21]: Acked

**Review by arkq — APPROVED** [review]

**Pull request merged**